src/file_pipe_log/pipe_builder.rs

```diff
@@ -143,24 +191,40 @@ impl<B: FileBuilder> DualPipesBuilder<B> {
             .unwrap();
         let (append, rewrite) = pool.join(
             || {
-                Self::recover_queue(
-                    LogQueue::Append,
-                    self.file_builder.clone(),
-                    self.cfg.recovery_mode,
-                    append_concurrency,
-                    self.cfg.recovery_read_block_size.0 as usize,
-                    &self.append_files,
-                )
+                if truncate_params.is_none()
+                    || truncate_params.as_ref().unwrap().queue.is_none()
+                    || truncate_params.as_ref().unwrap().queue.unwrap() == LogQueue::Append
+                {
+                    Self::recover_queue(
+                        LogQueue::Append,
+                        self.file_builder.clone(),
+                        self.cfg.recovery_mode,
+                        append_concurrency,
+                        self.cfg.recovery_read_block_size.0 as usize,
+                        &self.append_files,
+                        machine_builder,
+                    )
+                } else {
+                    Ok(None)
+                }
             },
             || {
-                Self::recover_queue(
-                    LogQueue::Rewrite,
-                    self.file_builder.clone(),
-                    self.cfg.recovery_mode,
-                    rewrite_concurrency,
-                    self.cfg.recovery_read_block_size.0 as usize,
-                    &self.rewrite_files,
-                )
+                if truncate_params.is_none()
+                    || truncate_params.as_ref().unwrap().queue.is_none()
+                    || truncate_params.as_ref().unwrap().queue.unwrap() == LogQueue::Append
```

Comment:
Add interface like `fn buildable(queue: LogQueue) -> bool` to MachineBuilder, instead of these very specific conditions.